lib/services/screen-builder-service.ts

```diff
@@ -27,11 +37,39 @@ export class ScreenBuilderService implements IScreenBuilderService {
 		return "add";
 	}
 
-	public prepareAndGeneratePrompt(generatorName: string, screenBuilderOptions?: IScreenBuilderOptions): IFuture<void> {
+	public prepareAndGeneratePrompt(generatorName: string, screenBuilderOptions?: IScreenBuilderOptions): IFuture<boolean> {
 		return (() => {
-			this.$logger.out("Please, wait while Screen Builder and its dependencies are being configured.");
-			this.promptGenerate(generatorName, screenBuilderOptions).wait();
-		}).future<void>()();
+			let scaffolderData = this.promptGenerate(generatorName, screenBuilderOptions).wait(),
+				scaffolderFutureResult = scaffolderData.future.wait(),
+				disableCommandHelpSuggestion = false;
+
+			if (scaffolderFutureResult === ScreenBuilderService.UPGRADE_ERROR_MESSAGE) {
+				this.$logger.error(ScreenBuilderService.UPGRADE_ERROR_MESSAGE_SHOWN_ON_THE_CONSOLE);
+				let shouldMigrate = this.$prompter.confirm('Do you want to upgrade your project now? Custom code changes might be lost.', () => false).wait();
+
+				if (shouldMigrate) {
+					let future = new Future<any>();
+					let callback = (err:Error, data:any) => {
+						if (err) {
+							let error = this.getErrorsRecursive(err).join('\n');
+							this.$logger.trace(`Screen Builder error while prompting: ${err.message}`);
```

Comment:
The message "  while prompting" is a bit too geeky. I think we can safely omit it.